Shopify/cli
# UI app console extend server func

### WHY are these changes introduced?
Make dev console extensible.  The motivation is to make it easy to extend the dev console so that we may build to a world where more dev tooling exist in this package.  That way it's easier for 3P developers to use the tooling we provide.

### WHAT is this pull request doing?

- Refactor the file structure of components to match [the React conventions that exist is web foundations](https://github.com/Shopify/web-foundations/blob/main/handbook/Best%20Practices/React/Organization.md)
- Move some markup and styles from the Dev Console component (now extensions) into a new Layout component.  We render the `<Routes/>` component inside this, so  new pages will get their styles.
- Add React router so we have a means to create new pages.

### How to test your changes?

Testing yarn dev:

1. Follow readme steps in `packages/ui-extensions-dev-console` and ensure dev console is reachable

Testing a production build:

1. Delete `packages/app/assets/dev-console`
2. `dev cd cli && yarn build`
3. `cd fixtures/app`
4. `yarn dev`
5. Open the dev console link the CLI outputs.  Does it render ok?

### Post-release steps

None

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
